azure/scope/machine.go

```diff
@@ -451,22 +451,14 @@ func (m *MachineScope) Role() string {
 	return infrav1.Node
 }
 
-// GetVMID returns the AzureMachine instance id by parsing Spec.FakeProviderID.
+// GetVMID returns the AzureMachine instance id by parsing the scope's providerID.
 func (m *MachineScope) GetVMID() string {
-	parsed, err := noderefutil.NewProviderID(m.ProviderID())
-	if err != nil {
-		return ""
-	}
-	return parsed.ID()
+	return utilazure.GetVMID(m.ProviderID())
 }
 
 // ProviderID returns the AzureMachine providerID from the spec.
 func (m *MachineScope) ProviderID() string {
-	parsed, err := noderefutil.NewProviderID(pointer.StringDeref(m.AzureMachine.Spec.ProviderID, ""))
-	if err != nil {
-		return ""
-	}
-	return parsed.String()
+	return utilazure.ProviderID(pointer.StringDeref(m.AzureMachine.Spec.ProviderID, ""))
```

Comment:
I wonder if we really need to do the parsing here or if we should just return the string as is. Since CAPI has gotten rid of Provider ID parsing altogether. And we're also not returning any error here if the parsing fails so this isn't adding any validation value.

Reply:
It passes it through regex validation, but doesn't actually return an error if it doesn't match (following the existing contract of these funcs). But you could argue the `ProviderID()` func should be a no-op. I'll try that.

Reply:
We do have a MachineScope test that expects its `ProviderID()` func to do validation here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/scope/machine_test.go#L198

IDK if the unit test was just written to match the (unneeded) parsing functionality or not, but it's a simple change if we're ok removing that test case.